### ISSUE TITLE
bug: typescript imports using windows paths

### ISSUE THREAD

**Issue by mrginglymus** [ionic-team/stencil#4543]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.0.1

### Current Behavior

On Windows, the `dist-custom-elements` target generates a `dist/components/index.d.ts` with some unescaped windows paths:

```ts
/* stencil custom elements */
export { Icon as VIcon } from '..\types\components\icons\icon';
export { defineCustomElement as defineCustomElementVIcon } from './v-icon';
```

### Expected Behavior

```ts
/* stencil custom elements */
export { Icon as VIcon } from '../types/components/icons/icon';
export { defineCustomElement as defineCustomElementVIcon } from './v-icon';
```

### System Info

```shell
System: node 16.20.0
    Platform: windows (10.0.19045)
   CPU Model: Intel(R) Xeon(R) Gold 5220 CPU @ 2.20GHz (8 cpus)
    Compiler: ...\playground\node_modules\.store\@stencil-core-npm-4.0.1-81248f31a7\node_modules\@stencil\core\compiler\stencil.js
       Build: 1687953951
     Stencil: 4.0.1
  TypeScript: 5.0.4
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.18.1
```

### Steps to Reproduce

On Windows, the `dist-custom-elements` target generates a `dist/components/index.d.ts` with some unescaped windows paths

### Code Reproduction URL

n/a

**Comment by alicewriteswrongs** [comment]
@mrginglymus unfortunately I don't have a windows machine handy for testing so I have to rely on our windows CI but I think I have a PR up (#4545) which will address the issue somewhat comprehensively. Is there any chance you could try running a dev build of Stencil to see if it fixes the issue for you?

If so, you can install it like this:

```sh
npm install --save-dev @stencil/core@4.0.1-dev.1688586331.51327eb
```

**Comment by mrginglymus** [comment]
Ah, that looks amazing, thanks so much for the quick and comprehensive response. I'll try and test this afternoon.

**Comment by mrginglymus** [comment]
Yes, that build has worked perfectly for me :)

**Comment by alicewriteswrongs** [comment]
Excellent, thanks for taking the time to test that out! I'll get that PR reviewed and merged in soon and it will be in the next release if all goes according to plan.

**Comment by alicewriteswrongs** [comment]
Just merged #4545, so it will be included in the next Stencil release. Thanks for reporting @mrginglymus!

**Comment by rwaskiewicz** [comment]
The fix for this issue has been released as a part of today's [Stencil v4.0.2 release](https://github.com/ionic-team/stencil/releases/tag/v4.0.2). As a result, I'm going to close this issue. Thanks!
